packages/expo-modules-core/ios/Core/Promise.swift

```diff
@@ -17,7 +17,16 @@ public struct Promise: AnyArgument {
     }
   }
 
-  public func resolve(_ value: Any? = nil) {
+  public func resolve(_ value: Any? = nil, appContext: AppContext? = nil) {
```

Comment:
Can we inject the `appContext` into the promise when creating it? The Promise object should be created by our API, so injecting the `appContext` there shouldn't be a problem.